test/test_playbooks/tasks/repository_sync.yml

```diff
@@ -0,0 +1,37 @@
+---
+- name: "Sync katello repository"
+  vars:
+    - organization_name: "Test Organization"
+    - product_name: "Test Product"
+    - repository: "Test Repository"
+    - synchronous: True
+  katello_sync:
+    username: "{{ foreman_username }}"
+    password: "{{ foreman_password }}"
+    server_url: "{{ foreman_server_url }}"
+    verify_ssl: "{{ foreman_verify_ssl }}"
+    organization: "{{ organization_name }}"
+    repository: "{{ repository }}"
+    product: "{{ product_name }}"
+    synchronous: "{{ synchronous }}"
+  register: result
+
+- fail:
+    msg: "Syncing repository failed! (expected_change: True)"
+  when: not result.changed
+
+- fail:
+    msg: "Syncing repository with synchronous: {{ synchronous }} took longer or shorter than expected!"
+  vars:
+    - synchronous: True
+  when: (synchronous and ((( result.changed.ended_at | replace(' UTC', '') | to_datetime ) - ( result.changed.started_at | replace(' UTC', '') | to_datetime )).total_seconds() < 2)) or
+        (not synchronous and result.changed.ended_at)
```

Comment:
I do not fully understand this condition.
If synchronous, then we fail if the task took less then 2 sec. OK.
If it is not, then what? `ended_at` is set by ansible, right (The module does not return anything meaningful)? Does this not fail every time?

Reply:
OK, then. Can you leave a comment there to avoid this confusion in the future?

Reply:
done :+1: 

thanks @mdellweg